lib/mainloop.c

```diff
@@ -280,14 +280,16 @@ main_loop_reload_config_apply(gpointer user_data)
   cfg_deinit(self->old_config);
   cfg_persist_config_move(self->old_config, self->new_config);
 
+  iv_task_register(&self->revert_config);
```

Comment:
Cosmetic, but this register cancel-register change is not needed anymore. What we need is to just add `main_loop_cancel_reenable_worker_jobs();` after `iv_task_register(&self->revert_config);`

Reply:
If we register `revert_config` after `cfg_init`, the list will be
```
  | invalidate_cached_time | _io_process_input | __iv_event_run_pending_events | _revert_config |
  ^
  | ivykis task list
```
The log message from `internal()` will be posted in `__iv_event_run_pending_events` and it also leads to a crash.

Therefore, I try to put `_revert_config` in the beginning of task list.

In this situation, the list can be illustrated as
```
  | _revert_config | invalidate_cached_time | _io_process_input | ... |
  ^
  | ivykis task list
```
After calling `_revert_config`, internal log message will be processed by reverted config.

Reply:
Thanks for the explanation. Marking as resolved.